lib/creator/yeoman/webpack-generator.js

```diff
@@ -1,138 +1,92 @@
-			console: false,
-			global: true,
-			process: true,
-			Buffer: true,
-			__filename: 'mock',
-			__dirname: 'mock',
-			setImmediate: true
-		}
-		this.configuration.dev.webpackOptions.performance = {
-			hints: "'warning'",
-			maxEntrypointSize: 400000,
-			maxAssetSize: 100000,
-			assetFilter: assetFilterFunction('js')
-		}
-		this.configuration.dev.webpackOptions.stats = 'errors-only'
-		this.configuration.dev.webpackOptions.amd = {
-			jQuery: true,
-			kQuery: false
-		}
-		this.configuration.dev.webpackOptions.bail = true;
-		this.configuration.dev.webpackOptions.cache = 'myCache'
-		this.configuration.dev.webpackOptions.profile = true
-		this.configuration.dev.webpackOptions.module = {
-			noParse: new RegExp(/jquery|lodash/),
-			rules: [
-				{
-					test: new RegExp('/\.jsx?$/'),
-					include: [
-						/* eslint-disable */
-						'path.resolve(__dirname, "app")'
-						/* eslint-enable */
-					],
-					exclude: [
-						/* eslint-disable */
-						'path.resolve(__dirname, ".." , "app", "demo-files")'
-						/* eslint-enable */
-					],
-					enforce: "'pre'",
-					loader: "'babel-loader'",
-					options: {
-						presets: ['es2015']
-					},
-				},
-				{
-					test: new RegExp('\\.html$'),
-					use: [
-						"'htmllint-loader'",
-						{
-							loader: "'html-loader'",
-							options: {
-								hello: "'world'"
+		let done = this.async();
+		let self = this;
+
+		this.configuration.config.webpackOptions.module = getDefaultLoaders();
+		this.configuration.config.webpackOptions.plugins = getDefaultPlugins();
+		this.prompt([
+			RawList('entryType', 'What kind of entry point do you want?', ['Array', 'Function', 'String', 'Object'])
+		]).then( (entryTypeAnswer) => {
+			// Ask different questions for entry points
+			entryQuestions(self, entryTypeAnswer).then(entryOptions => {
+				this.configuration.config.webpackOptions.entry = entryOptions;
+			}).then( () => {
+
+				this.prompt([
+					CheckList('outputType', 'What kind of output properties do you want?', [
+						'chunkFilename',
+						'filename',
+						'path'
+					])
+				]).then( (outputTypeAnswer) => {
+					// Ask different questions to output
+					OutputQuestions(self, outputTypeAnswer['outputType']).then(outputOptions => {
+						this.configuration.config.webpackOptions.output = outputOptions;
+					}).then( () => {
+						// Ask if the user wants to use extractPlugin
+						this.prompt([
+							Input('extractPlugin', 'If you got a css file, what\'s it named?')
+						]).then( (extractAnswer) => {
+							if(extractAnswer['extractPlugin'].length !== 0) {
+								this.configuration.config.webpackOptions.plugins.push(
+									'new ExtractTextPlugin(\'styles.css\')'
+								);
+								this.configuration.config.webpackOptions.module.rules.push({
+									test: new RegExp(/\.css$/),
+									use: `ExtractTextPlugin.extract({
+										fallback: 'style-loader',
+										use: 'css-loader'
+									})`
+								});
```

Comment:
Should we also ask any questions about their HTML? I see a lot of advanced projects using the HTML Webpack plugin for wiring up scripts. Perhaps it's worth also including here or is that overkill?